packages/eslint-plugin/src/rules/use-unknown-in-catch-callback-variable.ts

```diff
@@ -0,0 +1,403 @@
+                fix: (fixer): TSESLint.RuleFix =>
+                  fixer.insertTextAfter(catchVariableInner, ': [unknown]'),
+              },
+            ],
+          };
+        }
+        return {
+          node: catchVariableOuter,
+          suggest: [
+            {
+              messageId: 'wrongRestTypeAnnotationSuggestion',
+              fix: (fixer): TSESLint.RuleFix =>
+                fixer.replaceText(catchVariableTypeAnnotation, ': [unknown]'),
+            },
+          ],
+        };
+      }
+
+      return undefined;
+    }
+
+    return {
+      CallExpression(node): void {
+        // If there's no arguments, this rule doesn't apply.
+        if (node.arguments.length === 0) {
+          return;
+        }
+
+        if (!isPromiseCatchAccess(node.callee)) {
+          return;
+        }
+
+        const firstArgument = node.arguments[0];
+
+        // Deal with some special cases around spread element
+        if (firstArgument.type === AST_NODE_TYPES.SpreadElement) {
+          if (node.arguments.length === 1) {
+            if (checkWhetherShouldFlagSingleSpreadArg(firstArgument)) {
+              context.report({
+                node: firstArgument,
+                messageId: 'useUnknown',
+              });
+            }
+          } else {
+            if (checkWhetherShouldFlagMultipleSpreadArgs(node.arguments)) {
+              context.report({
+                node,
+                messageId: 'useUnknownSpreadArgs',
+              });
+            }
+          }
+          return;
+        }
+
+        // Normal cases.
+        if (checkWhetherShouldFlagArgument(firstArgument)) {
+          // We are now guaranteed to report, but we have a bit of work to do
+          // to determine exactly where, and whether we can fix it.
+          const overrides =
+            refineReportForNormalArgumentIfPossible(firstArgument);
+          context.report({
+            node: firstArgument,
+            messageId: 'useUnknown',
+            ...overrides,
+          });
+          return;
+        }
+      },
+    };
+  },
+});
+
+// TODO PR QUESTION - impl copied from prefer-find.
+// bump https://github.com/typescript-eslint/typescript-eslint/issues/8323 ?
+
+/**
+ * Answers whether the member expression looks like
+ * `x.memberName`, `x['memberName']`,
+ * or even `const mn = 'memberName'; x[mn]` (or optional variants thereof).
+ */
+function isStaticMemberAccessOfValue(
+  memberExpression:
+    | TSESTree.MemberExpressionComputedName
+    | TSESTree.MemberExpressionNonComputedName,
+  value: string,
+  scope?: Scope.Scope | undefined,
+): boolean {
+  if (!memberExpression.computed) {
+    // x.memberName case.
+    return memberExpression.property.name === value;
+  }
+
+  // x['memberName'] cases.
+  const staticValueResult = getStaticValue(memberExpression.property, scope);
+  return staticValueResult != null && value === staticValueResult.value;
+}
+
+function isParenlessArrowFunction(
```

Comment:
I'm pretty sure this function is not sound, but I also haven't figured out a way to break it. 

Is there a shared utility or better way to do this?